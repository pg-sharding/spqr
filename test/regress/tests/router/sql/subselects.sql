\c spqr-console

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;

CREATE KEY RANGE FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;

CREATE KEY RANGE FROM 201 ROUTE TO sh3 FOR DISTRIBUTION ds1;
CREATE KEY RANGE FROM 301 ROUTE TO sh4 FOR DISTRIBUTION ds1;

CREATE REFERENCE TABLE ref_rel_1;

ALTER DISTRIBUTION ds1 ATTACH RELATION table1 DISTRIBUTION KEY i;

\c regress

CREATE TABLE table1(i INT PRIMARY KEY);
CREATE TABLE ref_rel_1(i int, j int);

SELECT * FROM (SELECT * FROM table1 WHERE i = 1) AS a;
SELECT * FROM (SELECT * FROM (SELECT * FROM table1 WHERE i = 1) AS a) AS b;

DROP TABLE table1;
DROP TABLE ref_rel_1;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
