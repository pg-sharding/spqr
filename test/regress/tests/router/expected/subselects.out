\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 1
(1 row)

CREATE KEY RANGE FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 101
(1 row)

CREATE KEY RANGE FROM 201 ROUTE TO sh3 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 201
(1 row)

CREATE KEY RANGE FROM 301 ROUTE TO sh4 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 301
(1 row)

CREATE REFERENCE TABLE ref_rel_1;
         attach table          
-------------------------------
 relation name   -> ref_rel_1
 distribution id -> REPLICATED
(2 rows)

ALTER DISTRIBUTION ds1 ATTACH RELATION table1 DISTRIBUTION KEY i;
       attach table        
---------------------------
 relation name   -> table1
 distribution id -> ds1
(2 rows)

\c regress
CREATE TABLE table1(i INT PRIMARY KEY);
CREATE TABLE ref_rel_1(i int, j int);
SELECT * FROM (SELECT * FROM table1 WHERE i = 1) AS a;
 i 
---
(0 rows)

SELECT * FROM (SELECT * FROM (SELECT * FROM table1 WHERE i = 1) AS a) AS b;
 i 
---
(0 rows)

DROP TABLE table1;
DROP TABLE ref_rel_1;
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
       drop distribution       
-------------------------------
 distribution id -> REPLICATED
 distribution id -> ds1
(2 rows)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

